osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs

```diff
@@ -0,0 +1,376 @@
+                                                        ]
+                                                    },
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            ],
+                            null,
+                            [
+                                new Container
+                                {
+                                    RelativeSizeAxes = Axes.Both,
+                                    Padding = new MarginPadding
+                                    {
+                                        Horizontal = -WaveOverlayContainer.WIDTH_PADDING,
+                                    },
+                                    Children = new Drawable[]
+                                    {
+                                        new Box
+                                        {
+                                            RelativeSizeAxes = Axes.Both,
+                                            Colour = Color4Extensions.FromHex(@"28242d") // Temporary.
+                                        },
+                                        footerButtons = new FillFlowContainer
+                                        {
+                                            RelativeSizeAxes = Axes.Both,
+                                            Direction = FillDirection.Horizontal,
+                                            Padding = new MarginPadding(5),
+                                            Spacing = new Vector2(10),
+                                            Children = new Drawable[]
+                                            {
+                                                new PlaylistsReadyButton
+                                                {
+                                                    Anchor = Anchor.Centre,
+                                                    Origin = Anchor.Centre,
+                                                    RelativeSizeAxes = Axes.Y,
+                                                    Size = new Vector2(250, 1),
+                                                    Action = startPlay
+                                                }
+                                            }
+                                        },
+                                    }
+                                }
+                            ],
+                        }
+                    }
+                }
+            };
+
+            LoadComponent(userModsSelectOverlay = new RoomModSelectOverlay
+            {
+                SelectedMods = { BindTarget = userMods },
+                IsValidMod = _ => false
+            });
+
+            if (playlistItem.AllowedMods.Any())
+            {
+                footerButtons.Insert(0, new UserModSelectButton
```

Comment:
I'm guessing the goal here was to insert the button to the left of the "Start" button, but this doesn't seem to be working:

![2024-06-11 21 37 50@2x](https://github.com/ppy/osu/assets/191335/f233dbcf-e90e-4a01-a4d6-7047a10f3e9b)

Not sure if this is the best fix, but it works:

```diff
diff --git a/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs b/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs
index bff58dbb58..c0a0d4221b 100644
--- a/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs
+++ b/osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs
@@ -257,7 +257,7 @@ [new MatchChatDisplay(room) { RelativeSizeAxes = Axes.Both }]
 
             if (playlistItem.AllowedMods.Any())
             {
-                footerButtons.Insert(0, new UserModSelectButton
+                var userModSelectButton = new UserModSelectButton
                 {
                     Text = "Free mods",
                     Anchor = Anchor.Centre,
@@ -265,7 +265,10 @@ [new MatchChatDisplay(room) { RelativeSizeAxes = Axes.Both }]
                     RelativeSizeAxes = Axes.Y,
                     Size = new Vector2(250, 1),
                     Action = () => userModsSelectOverlay.Show(),
-                });
+                };
+
+                footerButtons.Add(userModSelectButton);
+                footerButtons.SetLayoutPosition(userModSelectButton, -1);
 
                 var rulesetInstance = rulesets.GetRuleset(playlistItem.RulesetID)!.CreateInstance();
                 var allowedMods = playlistItem.AllowedMods.Select(m => m.ToMod(rulesetInstance));

```

Reply:
> I'm guessing the goal here was to insert the button to the left of the "Start" button

It was, not sure how I didn't notice that not working. Especially so that I knew about this footgun from before.

> Not sure if this is the best fix, but it works:

`FlowContainer.Insert()` is actually a shorthand for the two operations you have in that patch there. Hence, https://github.com/ppy/osu/pull/28440/commits/5e002fbf9b8af739189bad1a12e4ab288375e640 suffices.

Reply:
Weird, I tried that exact patch and it crashed my test setup. Probably the `arm64` dynamic recompile "Program incorrect format" .NET issue which I just presumed to be me doing something wrong with the `Insert` call 🤷